src/directives/formly-form.test.js

```diff
@@ -952,5 +952,27 @@ describe('formly-form', () => {
       expect(listener).to.have.been.called
       expect(expression).to.have.been.called
     })
+
+    it(`should add watches on deep dive fields`, () => {
+      const formWithOptions = '<formly-form model="model" fields="fields" options="options"></formly-form>'
+      scope.model = {}
+      scope.options = {}
+
+      const deepLinkField = getNewField()
+      deepLinkField.key = 'foo.bar'
+      deepLinkField.watcher = {
+        listener: sinon.spy(),
+      }
+
+      scope.fields = [deepLinkField]
+      compileAndDigest(formWithOptions)
+      expect(compileAndDigest).to.not.throw()
+      expect(deepLinkField.watcher.listener).to.have.been.called
+      scope.model[deepLinkField.key] = 'brown'
```

Comment:
I am soo so sorry, but I just realized that this doesn't work. You'll want to do `scope.model.foo.bar = 'brown'` for this to work properly.